broker/src/main/java/io/camunda/zeebe/broker/system/configuration/DiskCfg.java

```diff
@@ -0,0 +1,129 @@
+  private static final String DISABLED_DISK_FREESPACE = "0%";
+  private static final Duration DEFAULT_DISK_USAGE_MONITORING_DELAY = Duration.ofSeconds(1);
+  private boolean enableMonitoring = DEFAULT_DISK_MONITORING_ENABLED;
+  private Duration monitoringInterval = DEFAULT_DISK_USAGE_MONITORING_DELAY;
+  private FreeSpaceCfg freeSpace = new FreeSpaceCfg();
+
+  @Override
+  public void init(final BrokerCfg globalConfig, final String brokerBase) {
+    freeSpace.init(globalConfig, brokerBase);
+
+    if (!enableMonitoring) {
+      LOG.info(
+          "Disk usage monitoring is disabled, setting required freespace to {}",
+          DISABLED_DISK_FREESPACE);
+      freeSpace.setReplication(DISABLED_DISK_FREESPACE);
+      freeSpace.setProcessing(DISABLED_DISK_FREESPACE);
+    }
+  }
+
+  public boolean isEnableMonitoring() {
+    return enableMonitoring;
+  }
+
+  public void setEnableMonitoring(final boolean enableMonitoring) {
+    this.enableMonitoring = enableMonitoring;
+  }
+
+  public Duration getMonitoringInterval() {
+    return monitoringInterval;
+  }
+
+  public void setMonitoringInterval(final Duration monitoringInterval) {
+    this.monitoringInterval = monitoringInterval;
+  }
+
+  public FreeSpaceCfg getFreeSpace() {
+    return freeSpace;
+  }
+
+  public void setFreeSpace(final FreeSpaceCfg freeSpace) {
+    this.freeSpace = freeSpace;
+  }
+
+  @Override
+  public String toString() {
+    return "DiskCfg{" + "enableMonitoring=" + enableMonitoring + ", freeSpace=" + freeSpace + '}';
+  }
+
+  public static class FreeSpaceCfg implements ConfigurationEntry {
+
+    private static final String DEFAULT_PROCESSING_FREESPACE = "2GB";
+    private static final String DEFAULT_REPLICATION_FREESPACE = "1GB";
+    private String processing = DEFAULT_PROCESSING_FREESPACE;
+    private String replication = DEFAULT_REPLICATION_FREESPACE;
+
+    public String getProcessing() {
+      return processing;
+    }
+
+    public void setProcessing(final String processing) {
+      this.processing = processing;
+    }
+
+    public String getReplication() {
+      return replication;
+    }
+
+    public void setReplication(final String replication) {
+      this.replication = replication;
+    }
+
+    public long getMinFreeSpaceForProcessing(final String dataDirectory) {
+      return parse(dataDirectory, processing);
+    }
+
+    public long getMinFreeSpaceForReplication(final String dataDirectory) {
+      return parse(dataDirectory, replication);
+    }
+
+    private long parse(final String dataDirectory, final String watermark) {
+      if (watermark.endsWith("%")) {
+        final NumberFormat defaultFormat = NumberFormat.getPercentInstance();
+        try {
+          final Number minFreeRatio = defaultFormat.parse(watermark);
+          final var directoryFile = new File(dataDirectory);
+          return Math.round(minFreeRatio.doubleValue() * directoryFile.getTotalSpace());
```

Comment:
> This is a behavior change that wasn't intended I think

I think not. Previously also we were translating percentage to absolute values at the beginning. See https://github.com/camunda/zeebe/blob/main/broker/src/main/java/io/camunda/zeebe/broker/system/monitoring/DiskSpaceUsageMonitorActor.java#L32 Required free space is calculated once when we start the monitor.

Also, for disk resizing using percentage is not useful. See #5280 In that case using absolute values would be recommended.

Reply:
Sorry my bad. Actually #5280 explains the issue that you mentioned. So resolving percentage every time we check would solve that issue for command. But it would not solve it for replication watermark. Replication watermark is checked in raft whenever the segment is created.

Reply:
Thinking about it more, I think using percentages are not a good idea. To configure percentages, we first calculated how much free space we need, then calculate the percentage based on that value and the allocated disk space. My recommendation would be to always use absolute values. So my suggestion is to do not accept % in the configuration.

Reply:
> I think not. Previously also we were translating percentage to absolute values at the beginning

> Sorry my bad. Actually #5280 explains the issue that you mentioned

When reviewing yesterday, I thought this worked previously but obviously it wasn't. So it's not a "behavior change that wasn't intended" :+1: 

I think that we should choose between two options:
1. Not supporting percentages at all
2. Supporting them, but always re-checking available space

First option would be easier of course. The only use case I can come up with for the second one is some sort of disk sharing where users wish to allocate say 50% of the disk to Zeebe and the rest to some other application(s). I'm not sure if that's really something that we want to support. 

Personally, I'd choose option two because it's not that much extra work (I hope :sweat_smile:) and we supported percentages previously.

Reply:
> Supporting them, but always re-checking available space

To do this for replication watermark, we would have to add one more i/o in the hot path of raft when creating a new segment.

Reply:
I think, code wise it is not much work to support both. But personally, I think percentages are not adding much value as we have seen with the old configuration.

Reply:
> To do this for replication watermark, we would have to add one more i/o in the hot path of raft when creating a new segment.

That convinced me that we shouldn't support percentages at all :+1:

Reply:
:+1: I will update the PR to remove support for percentages. We will still support it for the old config, but translate it to absolute value once onn startup.